src/electron/preview.tsx

```diff
@@ -0,0 +1,330 @@
+import * as HtmlSketchApp from '@brainly/html-sketchapp';
+import { HighlightArea } from './highlight-area';
+import { camelCase, omit } from 'lodash';
+import * as MobX from 'mobx';
+import * as MobXReact from 'mobx-react';
+import * as Path from 'path';
+import * as React from 'react';
+import * as ReactDom from 'react-dom';
+import * as SmoothscrollPolyfill from 'smoothscroll-polyfill';
+
+function main(): void {
+	SmoothscrollPolyfill.polyfill();
+
+	const store = MobX.observable.map({});
+	store.set('components', []);
+	store.set('tasks', []);
+
+	const highlight = new HighlightArea();
+
+	const connection = new WebSocket(`ws://${window.location.host}`);
```

Comment:
Did you think about using wss which would encrypt all the traffic between the server and the iFrame?

Reply:
Yes, but currently we do not have a good way to produce certificates on the fly. Any suggestions?

Reply:
Good point. We should rethink the effort vs security threat gain.